### PR TITLE
creating new DedUpSite

### PR DESCRIPTION
**Pull request by NithyashreeGS30**

### CODE REVIEW

File: www/src/models/mineralSite/MineralSite.ts [hunk 1]
```diff
@@ -146,7 +146,7 @@ export class DraftCreateMineralSite extends MineralSite {
   draftID: string;
 
   constructor({ draftID, ...rest }: { draftID: string } & MineralSiteConstructorArgs) {
-    super(rest);
+    super({ ...rest });
```
Comment (binh-vu) [inline www/src/models/mineralSite/MineralSite.ts:149]: change to just super(rest);

File: www/src/models/units/UnitStore.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { RStore, FetchResponse } from "gena-app";
+import { SERVER } from "env";
+
+export interface Unit {
+    id: string;
```
Comment (binh-vu) [inline www/src/models/units/UnitStore.tsx:5]: change the typing from string to `IRI` in `typing.ts`

File: www/src/models/units/UnitStore.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    name: string;
+}
+
+export class UnitStore extends RStore<string, Unit> {
+    constructor() {
+        super(`${SERVER}/api/v1/units`, undefined, false);
+    }
+
+    public getByURI(uri: string): Unit | undefined {
```
Comment (binh-vu) [inline www/src/models/units/UnitStore.tsx:14]: change string to `IRI`

File: www/src/models/units/UnitStore.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    public deserialize(obj: any): Unit {
+        return {
+            id: obj.uri,
+            name: obj.name,
+        };
+    }
+
+    protected normRemoteSuccessfulResponse(resp: any): FetchResponse {
+        return { items: resp.data, total: resp.total };
```
Comment (binh-vu) [inline www/src/models/units/UnitStore.tsx:34]: bug, use `total: resp.data.length` instead -- please update other stores country, state, commodity, and deposit type for me.

File: www/src/pages/home/DedupMineralSiteTable.tsx [hunk 5]
```diff
@@ -252,21 +255,30 @@ export const DedupMineralSiteTable: React.FC<DedupMineralSiteTableProps> = obser
...
+          </div>,
+          <Table<DedupMineralSite>
+            bordered={true}
+            size="small"
+            rowKey="id"
+            pagination={false}
+            columns={columns}
+            dataSource={Array.from(selectedDedupSiteIds).map((id) => dedupMineralSiteStore.get(id)!)}
+          />,
+        ]
```
Comment (binh-vu) [inline www/src/pages/home/DedupMineralSiteTable.tsx:271]: please rebase, old code

File: www/src/pages/home/NewMineralSite/NewMineralSiteModal.tsx [hunk 6]
```diff
@@ -0,0 +1,430 @@
...
+        stateOrProvinceStore.fetchAll().then(() => {
+            setStateOptions(Array.from(stateOrProvinceStore.records.values()).filter((state): state is StateOrProvince => state !== null).map((state: StateOrProvince) => ({ value: state.id, label: state.name, })));
+        });
+
+        depositTypeStore.fetchAll().then(() => {
+            setDepositTypeOptions(Array.from(depositTypeStore.records.values()).filter((type): type is DepositType => type !== null).map((type) => ({ value: type.id ?? "", label: type.name ?? "Unnamed", })));
+        });
+        unitStore.fetchAll().then(() => {
+            setUnitOptions(Array.from(unitStore.records.values()).filter((unit): unit is Unit => unit !== null).map((unit) => ({ value: unit.id ?? "", label: unit.name ?? "Unnamed", })));
+        });
```
Comment (binh-vu) [inline www/src/pages/home/NewMineralSite/NewMineralSiteModal.tsx:61]: don't do it here

File: www/src/pages/home/NewMineralSite/NewMineralSiteModal.tsx [hunk 7]
```diff
@@ -0,0 +1,430 @@
...
+        const generatedId = `record-${uuidv4()}`;
+        setRecordId(generatedId);
+        form.setFieldsValue({ recordId: generatedId });
+    }, [commodityStore, countryStore, stateOrProvinceStore, form, visible]);
+
+    const handleSave = async (values: any) => {
+        try {
+            const currentUser = userStore.getCurrentUser()?.name;
+            const createdBy = `https://minmod.isi.edu/users/u/${currentUser}`;
```
Comment (binh-vu) [inline www/src/pages/home/NewMineralSite/NewMineralSiteModal.tsx:70]: we do not hard code createdBy any more

File: www/src/pages/home/NewMineralSite/NewMineralSiteModal.tsx [hunk 8]
```diff
@@ -0,0 +1,430 @@
...
+                reference: [reference],
+                sameAs: [],
+                gradeTonnage: {
+                    [commodity1 as string]: new GradeTonnage({
+                        commodity: commodity1,
+                        totalTonnage: values.tonnage || 0,
+                        totalGrade: values.grade || 0,
+                    }),
+                },
+                mineralInventory: [],
```
Comment (binh-vu) [inline www/src/pages/home/NewMineralSite/NewMineralSiteModal.tsx:146]: populate the mineralInventory field with data

Comment (binh-vu) [reply www/src/pages/home/NewMineralSite/NewMineralSiteModal.tsx:146]: gradeTonnage is temporary and will be discard if there is no mineral inventory

**Review by binh-vu — CHANGES_REQUESTED** [review]

**Comment by binh-vu** [comment]
rank default is U, type default is unspecified.

File: www/src/pages/home/NewMineralSite/NewMineralSiteModal.tsx [hunk 9]
```diff
@@ -0,0 +1,430 @@
...
+
+                    }),
+                ]
+                : [];
+
+            const statesOrProvinces = values.stateorprovince
+                ? [
+                    new CandidateEntity({
+                        observedName: values.stateorprovince,
+                        source: values.stateorprovince,
```
Comment (binh-vu) [inline www/src/pages/home/NewMineralSite/NewMineralSiteModal.tsx:93]: incorrect source, use user url

File: www/src/pages/home/NewMineralSite/NewMineralSiteModal.tsx (hunk 6 above)
Comment (NithyashreeGS30) [reply www/src/pages/home/NewMineralSite/NewMineralSiteModal.tsx:61]: where do you want me to move it to ?

**Pull request closed without merging**